Packs/CiscoEmailSecurity/Integrations/CiscoEmailSecurity/CiscoEmailSecurity.py

```diff
@@ -0,0 +1,629 @@
+            'Accept': 'application/json'
+        }
+        data = {
+            "data":
+                {
+                    "userName": base64.b64encode(self.username.encode('ascii')).decode('utf-8'),
+                    "passphrase": base64.b64encode(self.password.encode('ascii')).decode('utf-8')
+                }
+        }
+        output = self.http_request('POST', '/sma/api/v2.0/login', json_data=data, headers=headers)
+        jwt_token = output.get('data').get('jwtToken')
+        return jwt_token
+
+    def list_report(self, url_suffix) -> Dict[str, Any]:
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix,
+        )
+
+    def list_messages(self, url_suffix) -> Dict[str, Any]:
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_get_message_details(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_get_dlp_details(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_get_amp_details(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_get_url_details(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_spam_quarantine(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_quarantine_get_details(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_delete_quarantine_messages(self, url_suffix, request_body):
+        return self.http_request(
+            method='DELETE',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+    def list_release_quarantine_messages(self, url_suffix, request_body):
+        return self.http_request(
+            method='POST',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+    def list_entries_get(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_entries_add(self, url_suffix, request_body):
+        return self.http_request(
+            method='POST',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+    def list_entries_delete(self, url_suffix, request_body):
+        return self.http_request(
+            method='DELETE',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+
+def test_module(client: Client) -> str:
+    suffix_url = "/sma/api/v2.0/message-tracking/messages?startDate=2018-01-01T00:00:00.000Z&" \
```

Comment:
why are we mentioning dates? what happens if the date doesn't exist in the system?

Reply:
The dates are a required parameter for all API queries

Reply:
hat happens if the date doesn't exist in the system?